### PR TITLE
feat(Button): add new button design and introduce emphasis property [#Button 2]

### PR DESCRIPTION
**Pull request by julianiff**
New Button styles with current Main Branch Merged

### CODE REVIEW

**Comment by julianiff** [comment]
@imoutaharik @thisguyscode I opened up a new branch with the merge of the main to easily go back if smth doesn't fit with the merges from the main.

**Comment by julianiff** [comment]
see for terriffic button styling update https://github.com/Frontify/clarify/pull/9885

File: .sonarcloud.properties [hunk 1]
```diff
@@ -1,3 +1,3 @@
 sonar.cpd.exclusions=src/foundation/Icon/IconsMap.tsx
-sonar.exclusions=src/foundation/Icon/IconsMap.tsx
+sonar.exclusions=src/foundation/Icon/IconsMap.tsx,src/components/Button/ButtonClasses.tsx
```
Comment (imoutaharik) [inline .sonarcloud.properties:2]: Nice!!! This is exactly what I was looking for as we had a serious issue of duplication on that file. 🙌🏼

File: src/components/Button/Button.stories.tsx [hunk 2]
```diff
@@ -2,81 +2,83 @@
...
 export const withIcon = ButtonTemplate.bind({});
-withIcon.args = {
-    disabled: false,
-    icon: <IconIcon />,
-    size: ButtonSize.Medium,
-    solid: true,
-    style: ButtonStyle.Primary,
-    type: ButtonType.Button,
-};
+withIcon.args = { ...defaultArgs, hideLabel: true, icon: iconsMap[IconEnum.Icons] };
```
Comment (imoutaharik) [inline src/components/Button/Button.stories.tsx:67]: it should be IconEnum.Icon as IconEnum.Icons (in plural) is the old one.

File: src/components/Button/Button.stories.tsx [hunk 3]
```diff
@@ -2,81 +2,83 @@
...
 withIconAndLabel.args = {
-    children: 'Button Text',
-    disabled: false,
-    icon: <IconIcon />,
-    size: ButtonSize.Medium,
-    solid: true,
-    style: ButtonStyle.Primary,
-    type: ButtonType.Button,
+    ...defaultArgs,
+    icon: iconsMap[IconEnum.ColorScale],
```
Comment (imoutaharik) [inline src/components/Button/Button.stories.tsx:82]: Same here, ColorScale is now ColorFan.

File: src/components/ButtonGroup/ButtonGroup.stories.tsx [hunk 4]
```diff
@@ -12,22 +12,34 @@ export default {
...
     args: {
         size: ButtonSize.Medium,
+        style: ButtonStyle.Default,
+        emphasis: ButtonEmphasis.Default,
     },
 } as Meta<ButtonGroupProps>;
 
-export const ButtonGroup: Story<ButtonGroupProps> = (args: ButtonGroupProps) => (
+export const ButtonGroup = ({ ...args }) => (
     <ButtonGroupComponent size={args.size}>
```
Comment (imoutaharik) [inline src/components/ButtonGroup/ButtonGroup.stories.tsx:35]: Same here icons should be:
IconAddSimple -> IconPlus
IconCheck -> IconCheckMark
IconReject -> IconCross

**Comment by julianiff** [comment]
@imoutaharik I adjusted your inputs, thx and refactored more of the button. Looking forward for your input there.

**Comment by julianiff** [comment]
@imoutaharik I updated the Button Component to be non-breaking (readded the properties with deprecated). When time permits, looking forward to your review :)

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**